src/audio/mfcc/mfcc_hifi3.c

```diff
@@ -121,7 +121,7 @@ void mfcc_fill_prev_samples(struct mfcc_buffer *buf, int16_t *prev_data,
 
 	buf->s_avail -= prev_data_length;
 	buf->s_free += prev_data_length;
-	buf->r_ptr = in;
+	buf->r_ptr = (void *)in; /* int16_t pointer but direct cast is not possible */
```

Comment:
is ```in``` the wrong ptr type, do we need to look at other changes here ?

Reply:
It's up to @andrula-song to confirm but in the patch she used ae_int32 type to load two ae_int16 types, that's in comments a few lines above this typecast. I was wondering too if this can be correct but it appears so.

Reply:
ok, intrinsic types are non portable. Lets revisist as we do need to support AVX here (and the API would need to become more generic).

Reply:
@singalsu  @lgirdwood , I used to use AE_LA16X4_XC to load a 16bit * 4 data, but I found the result is not right, I think it is a bug of xtensa. So I use AE_L32_XC to load a 32bit data(16bit * 2) to replace twice AE_L16_XC, and force the datatype to ae_int16 to copy the last 16 bit if the input data size is odd.